packages/core/helper-plugin/src/components/GenericInput.tsx

```diff
@@ -22,13 +23,61 @@ import {
 } from '@strapi/design-system';
 import { Eye, EyeStriked } from '@strapi/icons';
 import formatISO from 'date-fns/formatISO';
-import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 
-import { useFieldHint } from '../hooks/useFieldHint';
+import { FieldSchema, useFieldHint } from '../hooks/useFieldHint';
 import { useFocusInputField } from '../hooks/useFocusInputField';
 import { pxToRem } from '../utils/pxToRem';
 
+import type { InputType, TranslationMessage } from '../types';
+import type { Attribute } from '@strapi/strapi';
+
+interface InputOption {
+  metadatas: {
+    intlLabel: TranslationMessage;
+    disabled: boolean;
+    hidden: boolean;
+  };
+  key: string;
+  value: string;
+}
+
+interface CustomInputProps<TAttribute extends Attribute.Any>
+  extends Omit<GenericInputProps<TAttribute>, 'customInputs'> {
+  ref?: React.Ref<HTMLElement>;
+  hint?: string | React.JSX.Element | (string | React.JSX.Element)[];
+}
+
+interface GenericInputProps<TAttribute extends Attribute.Any = Attribute.Any> {
```

Comment:
@markkaylor when we (eventually) refactor this, we can be more clever about it and infer the props using the attribute type to understand which component should be rendered and when etc.